rust_src/src/fns.rs

```diff
@@ -306,4 +308,39 @@ pub extern "C" fn internal_equal_misc(
     }
 }
 
+/// Return list of 1 minute, 5 minute and 15 minute load averages.
+///
+/// Each of the three load averages is multiplied by 100, then converted
+/// to integer.
+///
+/// When USE-FLOATS is non-nil, floats will be used instead of integers.
+/// These floats are not multiplied by 100.
+///
+/// If the 5-minute or 15-minute load averages are not available, return a
+/// shortened list, containing only those averages which are available.
+///
+/// An error is thrown if the load average can't be obtained.  In some
+/// cases making it work would require Emacs being installed setuid or
+/// setgid so that it can read kernel information, and that usually isn't
+/// advisable.
+#[lisp_fn(min = "0")]
+pub fn load_average(use_floats: LispObject) -> LispObject {
+    let mut load_avg: [libc::c_double; 3] = [0.0, 0.0, 0.0];
+    let loads = unsafe { libc::getloadavg(load_avg.as_mut_ptr(), 3) };
+
+    if loads < 0 {
+        error!("load-average not implemented for this operating system");
+    }
+
+    (0..loads as usize).rev().fold(Qnil, |acc, i| {
```

Comment:
`From<Vec<LispObject>> for LispObject` is implemented, so I think you can return `loads` as-is (or map whatever float converter) rather than manually constructing a Lisp list.

Reply:
So it is. I've updated it so we generate a `Vec<LispObject>` and we generate the list out of that with `From`.